From 2f07fdd5d34673f856b8f8401ab8447000254aaa Mon Sep 17 00:00:00 2001
From: David Rosca <nowrep@gmail.com>
Date: Thu, 14 Jan 2016 18:54:57 +0100
Subject: [PATCH] Fix shrinking panel on top/left/right edge

Use the new computed value for configDialog position.
Fix regression from 8ba3633227

REVIEW: 126743
BUG: 357835
FIXED-IN: 5.5.4
---
 .../contents/configuration/panelconfiguration/SizeHandle.qml        | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/desktoppackage/contents/configuration/panelconfiguration/SizeHandle.qml b/desktoppackage/contents/configuration/panelconfiguration/SizeHandle.qml
index 5e6185c..753b6f4 100644
--- a/desktoppackage/contents/configuration/panelconfiguration/SizeHandle.qml
+++ b/desktoppackage/contents/configuration/panelconfiguration/SizeHandle.qml
@@ -45,7 +45,7 @@ PlasmaComponents.Button {
             switch (panel.location) {
             case PlasmaCore.Types.TopEdge:
                 var y = Math.min(panel.screen.geometry.y + panel.screen.geometry.height/2, mouse.screenY - mapToItem(dialogRoot, 0, startMouseY).y);
-                var thickness = Math.max(units.gridUnit, configDialog.y - panel.y);
+                var thickness = Math.max(units.gridUnit, y - panel.y);
 
                 if (thickness % 2 != 0) {
                     if (mouse.y > startMouseY) {
@@ -62,7 +62,7 @@ PlasmaComponents.Button {
                 break;
             case PlasmaCore.Types.LeftEdge:
                 var x = Math.min(panel.screen.geometry.x + panel.screen.geometry.width/2, mouse.screenX - mapToItem(dialogRoot, startMouseX, 0).x);
-                var thickness = Math.max(units.gridUnit, configDialog.x - panel.x);
+                var thickness = Math.max(units.gridUnit, x - panel.x);
 
                 if (thickness % 2 != 0) {
                     if (mouse.x > startMouseX) {
@@ -79,7 +79,7 @@ PlasmaComponents.Button {
                 break;
             case PlasmaCore.Types.RightEdge:
                 var x = Math.max(panel.screen.geometry.x + panel.screen.geometry.width/2, mouse.screenX - mapToItem(dialogRoot, startMouseX, 0).x);
-                var thickness = Math.max(units.gridUnit, panel.screen.geometry.x + panel.screen.geometry.width - (configDialog.x + configDialog.width));
+                var thickness = Math.max(units.gridUnit, panel.screen.geometry.x + panel.screen.geometry.width - (x + configDialog.width));
 
                 if (thickness % 2 != 0) {
                     if (mouse.x > startMouseX) {
-- 
2.7.0.rc3

